test/unit/core/detail/reflection_test.cpp

```diff
@@ -83,7 +85,93 @@ class reflection : public ::testing::Test
 
 TYPED_TEST_CASE(reflection, reflection_types);
 
+TYPED_TEST(reflection, size)
+{
+    EXPECT_EQ(detail::get_display_name_size_v<TypeParam>, this->expected_name().size());
+}
+
 TYPED_TEST(reflection, name)
 {
     EXPECT_EQ(detail::get_display_name_v<TypeParam>.string(), this->expected_name());
 }
+
+namespace seqan3::detail
+{
+template <typename type>
+struct pretty_lambda
+{
+    static constexpr auto as_string = [] ()
+    {
+        return std::string{__PRETTY_FUNCTION__};
+    };
+};
+} // namespace seqan3::detail
+
+TEST(pretty_lambda, as_string)
+{
+    // NOTE: remove this if it is to much maintenance work
+    std::string int_ = seqan3::detail::pretty_lambda<int>::as_string();
+#if defined(__clang__)
+    EXPECT_EQ(int_, "auto seqan3::detail::pretty_lambda<int>::(anonymous class)::operator()() const [type = int]");
+#elif defined(__GNUC__) && __GNUC__ >= 8
+    EXPECT_EQ(int_, "seqan3::detail::pretty_lambda<int>::<lambda()>");
+#elif defined(__GNUC__)
+    EXPECT_EQ(int_, "seqan3::detail::pretty_lambda<type>::<lambda()> [with type = int]");
+#endif
+
+    std::string foo_bar_char = seqan3::detail::pretty_lambda<foo::bar<char>>::as_string();
+#if defined(__clang__)
+    EXPECT_EQ(foo_bar_char, "auto seqan3::detail::pretty_lambda<foo::bar<char> >::(anonymous class)::operator()() const [type = foo::bar<char>]");
+#elif defined(__GNUC__) && __GNUC__ >= 8
+    EXPECT_EQ(foo_bar_char, "seqan3::detail::pretty_lambda<foo::bar<char> >::<lambda()>");
+#elif defined(__GNUC__)
+    EXPECT_EQ(foo_bar_char, "seqan3::detail::pretty_lambda<type>::<lambda()> [with type = foo::bar<char>]");
+#endif
+}
```

Comment:
You are not testing the library here, but instead local stuff from the tests... why?

Reply:
It ensures that the format of the underlying used method stays the same. Furthermore, It took me quite some time to figure out, what the output of `__PRETTY_FUNCTION__` is.

Reply:
But the tests should always test the library and not "something similar" also implemented in the test. If the code tested in the test is very similar to the actual library code it shouldn't be necessary, if it is substantially different, then chances are big the code will generate false positive and/or false negative test failures.
E.g.: doing a full string comparison is much stricter than our actual library code, many changes to the string may be non-breaking.

Reply:
I think we have a different feeling about test cases.

The thing is that in this case we base our functionality on some undocumented behaviour. The only way to ensure the behaviour is to actually test it and react to any change (even if it's stricter). This by design encourages false positives/negatives.

But I can remove it altogether and just leave 

```patch

+TEST(pretty_function, as_string)
+{
+    // an example how to extract the type
+    auto type_name = [](std::string pretty)
+    {
+        auto start = pretty.find("type = ") + std::string("type = ").size();
+        auto end = pretty.rfind("]");
+
+        return pretty.substr(start, end - start);
+    };
+
+    std::string int_ = seqan3::detail::pretty_function<int>::as_string();
+    EXPECT_EQ(type_name(int_), "int");
+
+    std::string foo_bar_char = seqan3::detail::pretty_function<foo::bar<char>>::as_string();
+    EXPECT_EQ(type_name(foo_bar_char), "foo::bar<char>");
+}
```